cornac/eval_methods/base_method.py

```diff
@@ -479,6 +479,14 @@ def _build_datasets(self, train_data, test_data, val_data=None):
         self.train_set.total_items = self.total_items
 
     def _build_modalities(self):
+        for modality in [self.sentiment, self.user_text, self.item_text]:
+            if modality is None or not isinstance(modality, (SentimentModality, ReviewModality)):
```

Comment:
Why do we need to include ReviewModality here?

Reply:
We can just add `uid_map, iid_map, dok_matrix` into the `build()` function of the below two loops

Reply:
This needs to be changed, otherwise, the `build()` function will be called twice for `TextModality`